### PR TITLE
Fixed unassigned hours issue

### PR DESCRIPTION
**Pull request by Sneha1b**
# Description
Currently, the time logged in the Other category tasks/project would not get reflected in the unassigned hours category in the user profile page, volunteering times tab

Fixes # (bug list priority high/medium/low x.y.z)
time logged in the Other category tasks/project gets reflected in the unassigned hours category in the user profile page, volunteering times tab

## Main changes explained:
Modified the updateHoursByCategory and editHoursByCategory function in the TimeEntryForm so that the userProfile categoryByHours gets updated correctly.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log in as owner user 
4. Create a project with category unspecified/Other
5. Create a task with category Other and assign it to a volunteer
6. Log in as the volunteer with assigned task and log time entry
7. Check if the time gets reflected in the 

## Screenshots or videos of changes:
Before
Screen Shot 2023-06-01 at 3 14 45 PM
After 
Screen Shot 2023-06-01 at 3 15 01 PM

### CODE REVIEW

**Review by 20chen-7 — CHANGES_REQUESTED** [review]
I tried as an admin user and volunteer user for test, but the committed hours could not be able to count as unassigned hrs.  
Screenshot 2023-06-05 at 10 16 10 AM
Screenshot 2023-06-05 at 10 12 26 AM (2)

**Review by leonzh2k — COMMENTED** [review]
Here's what I did to test the fix.
1. Started up the app on localhost
2. logged into an admin account and created task w/ category "Other" and assigned it to a volunteer account
image
3. logged into that volunteer account and logged tangible time to that task:
image
4. Checked the unassigned volunteer time: 
image

As you can see, it doesn't look like it updated.

**Review by stevenpaalz — CHANGES_REQUESTED** [review]
I am seeing some strange behavior on this. I am not able to recreate the intended functionality. Upon logging time under a task in the other category, tangible hours this week increases, but unassigned category hours does not.

Screenshot 2023-06-05 at 6 53 58 PM

However, if I log time under the project that contains the task with the other category, the logged time does appear under the unassigned category hours.

Screenshot 2023-06-05 at 7 11 18 PM

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx [hunk 1]
```diff
@@ -312,7 +312,7 @@ const TimeEntryForm = props => {
         : foundTask.classification.toLowerCase();
```
Comment (nathanah) [inline src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx:312]:
```suggestion
        : foundTask.category.toLowerCase();
```
Getting "TypeError: Cannot read properties of undefined (reading 'toLowerCase')". This appears to remove the error and possibly fix the issues others have been having.

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx [hunk 2]
```diff
@@ -368,7 +368,7 @@ const TimeEntryForm = props => {
     category = foundProject
       ? foundProject.category.toLowerCase()
       : foundTask?.classification.toLowerCase();
```
Comment (nathanah) [inline src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx:370]:
```suggestion
      : foundTask?.category.toLowerCase();
```
Same as other comment.

**Review by nathanah — CHANGES_REQUESTED** [review]

**Review by yufeiStella — CHANGES_REQUESTED** [review]
Once I assign a task to a volunteer while logged in as the admin, there is no change in the volunteer's allocated time on their profile.
Before:
before
Then assign a task to the volunteer:
Screen Shot 2023-06-09 at 22 12 24
After(while nothing changed): 
Screen Shot 2023-06-09 at 22 14 58

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx (hunk 1 above)
Comment (Sneha1b) [reply src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx]: On it. changed this

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx (hunk 2 above)
Comment (Sneha1b) [reply src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx]: Done

**Review by stevenpaalz — APPROVED** [review]
With the latest update, I can confirm that this is working with logging time under a task with the category other as well as a project under the category other. Note that it only works for tangible time.


Screenshot 2023-06-14 at 5 57 03 PM

**Review by leonzh2k — APPROVED** [review]
Works when adding tangible time but not when adding intangible time (not sure if this is intended)

**Review by bienzguoa — APPROVED** [review]
I added both tangible and intangible hours, it works as intended only for tangible hours. Not sure if the "Total Unassigned Category Hours" only means for tangible hours.
Screenshot 2023-06-17 at 6 56 22 PM
Screenshot 2023-06-17 at 7 02 02 PM

**Review by eduhorta — APPROVED** [review]
It is working as intended. Great job! 👍

Volunteer account before adding intangible time entry:
image

Time entry added:
image

After:
image

**Review by cvtqx — APPROVED** [review]
1. I created a task with Other category and assigned it to a volunteer:

Screen Shot 2023-06-22 at 4 15 59 PM

2. In volunteer account I logged time for that task as tangible time and it was reflected in the report:

Screen Shot 2023-06-22 at 4 13 54 PM

3. Like others mentioned, when logging the time for the same task is as intangible it is not reflected in the 'total unassigned category hours'. However, it is not mentioned in this PR, so it might be out of the scope. Hence approving.

Screen Shot 2023-06-22 at 4 20 00 PM

**Review by 20chen-7 — APPROVED** [review]
Looks like intended.
Screenshot 2023-07-07 at 10 11 08 AM

**Review by yufeiStella — APPROVED** [review]
Now it works perfectly!

**Pull request merged**